subsys/bluetooth/mesh/proxy.c

```diff
@@ -1132,7 +1132,7 @@ static int sub_count(void)
 
 static s32_t gatt_proxy_advertise(struct bt_mesh_subnet *sub)
 {
-	s32_t remaining = K_FOREVER;
+	s32_t remaining = (s32_t)K_TICKS_FOREVER;
```

Comment:
This doesn't seem right. `remaining` indicates a number of milliseconds but you're assigning to it a unit of ticks. Either hard-code `-1`, add a custom define, or introduce a system-wide `K_MS_FOREVER` define (or similar).

Reply:
I agree. This raises the question on whether we actually need a system-wide `K_MS_FOREVER` macro or we should leave it to the individual subsystems. @andyross and @nordic-krch thoughts?

Reply:
I'm sticking with hardcoding -1.

I did a test and `K_TIMEOUT_EQ(K_MSEC(-1), K_FOREVER);`, results in a true expression. But `K_MSEC` calls `k_ms_to_ticks_ceil32` and I'm not sure if that will always be the case.

Reply:
Using SYS_FOREVER_MS now that it has been added.